druid-shell/src/platform/windows/mod.rs

```diff
@@ -26,3 +26,194 @@ pub mod runloop;
 mod timers;
 pub mod util;
 pub mod window;
+
+// from winapi::um::d2d1;
+// need to move from one type to another?
+//
+// https://docs.microsoft.com/en-us/windows/win32/direct2d/render-targets-overview
+// ID2D1RenderTarget is the interface. The other resources inherit from it.
+//
+// A Render Target creates resources for drawing and performs drawing operations.
+//
+// - ID2D1HwndRenderTarget objects render content to a window.
+// - ID2D1DCRenderTarget objects render to a GDI device context.
+// - bitmap render target objects render to off-screen bitmap.
+// - DXGI render target objects render to  a DXGI surface for use with Direct3D.
+//
+// https://docs.microsoft.com/en-us/windows/win32/direct2d/devices-and-device-contexts
+// A Device Context, ID2D1DeviceContext, is used for windows 8 and higher. Render Target
+// is used for windows 7 and lower.
+//
+// Basically, go from HwndRenderTarget or DxgiSurfaceRenderTarget (2d or 3d) to a Device Context.
+// Go back up for particular needs. Move up and down using query_interface.
+use piet_common::d2d::{D2DFactory, DeviceContext};
+use std::fmt::{Debug, Display, Formatter};
+use winapi::Interface;
+use winapi::shared::windef::HWND;
+use winapi::shared::winerror::{HRESULT, SUCCEEDED};
+use winapi::um::d2d1::{D2D1_HWND_RENDER_TARGET_PROPERTIES, D2D1_RENDER_TARGET_PROPERTIES,
+                       D2D1_SIZE_U, ID2D1HwndRenderTarget, ID2D1RenderTarget};
+use winapi::um::d2d1_1::ID2D1DeviceContext;
+use winapi::um::dcommon::D2D1_PIXEL_FORMAT;
+use wio::com::ComPtr;
+
+// TODO make newtypes
+// Can probably follow the pattern of direct2d crate
+// e.g. https://github.com/Connicpu/direct2d-rs/blob/v0.1.2/src/render_target/hwnd.rs
+//
+// error and wrap, copy from d2d.rs
+//
+// for creating rendertarget, use
+// https://github.com/hwchen/piet/blob/raw_d2d/piet-direct2d/src/d2d.rs#L165 as example, but fill
+// in properties etc.
+
+#[derive(Clone)]
+pub struct HwndRenderTarget {
+    ptr: ComPtr<ID2D1HwndRenderTarget>
+}
+
+impl HwndRenderTarget {
+    pub fn create<'a>(factory: &'a D2DFactory, hwnd: HWND, width: u32, height: u32) -> Result<Self, Error> {
+        // hardcode
+        // - RenderTargetType::Default
+        // - AlphaMode::Unknown
+        let rt_props = DEFAULT_PROPS;
+        let mut hwnd_props = DEFAULT_HWND_PROPS;
```

Comment:
Pretty sure you need to set the hwnd here. I think it might make sense to write out the `D2D1_HWND_RENDER_TARGET_PROPERTIES { }` struct rather than setting fields, it means there's conscious thought to set all the fields.

Seeing the same pattern below, I think I know what's going on, a builder pattern is nice when you're exposing an API, but when the struct being built is in the same module as the use, it's less code (and imho simpler) just to write it as a struct.